userspace/libgolf2/digest_syscalls.h

```diff
@@ -0,0 +1,49 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TOCK_DIGEST_H
+#define TOCK_DIGEST_H
+
+#include <stdlib.h>
+
+// TODO These need to be standardized
+#define H1B_DRIVER_DIGEST 0x40003
+
+// command() type ids
+#define TOCK_DIGEST_CMD_INITIALIZE 0
+#define TOCK_DIGEST_CMD_UPDATE 1
+#define TOCK_DIGEST_CMD_FINALIZE 2
+
+// allow() type ids
+#define TOCK_DIGEST_ALLOW_INPUT 0
+#define TOCK_DIGEST_ALLOW_OUTPUT 1
+
+typedef enum TockDigestMode {
+  DIGEST_MODE_SHA1 = 0,
+  DIGEST_MODE_SHA256 = 1,
+  DIGEST_MODE_SHA256_HMAC = 2,
+} TockDigestMode;
+
+// TODO: Should be const, but currently not allowed by kernel
```

Comment:
You can be const-correct in this API and non-const-correct in the syscall API, as long as the syscall driver does not actually modify the buffer.

Reply:
The issue is that if you pass a const buffer, it could be in RO memory (flash); the current syscall implementations only allow you to allow() buffers that are in RAM. So the const is removed to ensure you don't try passing a const buffer.

Reply:
Oh right, I keep forgetting about that.